### DISCUSSION TITLE
Choose different values of the variable nyear (or nmonth, ndays) depending on the currently running model year

### DISCUSSION THREAD

**Discussion by nwieters** [esm-tools/esm_tools#985] [ESM-Tools support] [answered]
How to set up a runscript that allows to choose different values of the variable `nyear` (or `nmonth`, `ndays`) depending on the currently running model year.

For example:

Choose 
nyear = 10 if year 2010
and 
nyear = 5 if year >= 2010

**Comment by JanStreffing** [comment]
I think this should be possible with a normal choose block. https://esm-tools.readthedocs.io/en/latest/yaml.html?highlight=choose#switches-choose

**Reply by nwieters** [reply]
Hi @JanStreffing , thank you for your tipp. I asked this here for another user and put it here, because this might be of interest for others as well.

I will try out your suggestion with a choose block and then write a runscripts-example snippet as an answer.

**Comment by nwieters** [comment, marked as answer]
Choosing the value for runscript variables like `nday`, `nmonth` or `nyear` is not possible. These variables are immutable and can therefore not be changed from the runscript. 
In order to run a simulation from 1.1.2000 to 31.12.2014 with `nyear: 5` and not ignoring the final date 31.12.2014, an easy solution would be to split the simulation into two. The first one running from 1.1.2000 to 31.12.2010 with `nyear: 5` and the second run from 1.1.2014 to 31.12.2014 with `nyear: 4`.
This is a solution that most user propably do already.
By keeping `nyear: 5` the simulation will ignore the final date (31.12.2014) and run for the full chunk length of `nyear: 5` (until 31.12.2015).
